### PR TITLE
docs: add docs for impression data

### PR DESCRIPTION
**Pull request by FredrikOseberg**
* Add documentation that details the new impression data feature

### CODE REVIEW

File: website/docs/advanced/impression_data.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+unleash.on("impression", (event) => {
+  // Capture the event here and pass it internal data lake or analytics provider
+  console.log(event);
+})
+```
+
+This will allow you to capture an event whenever a call is done to isEnabled or getVariant, capturing the context of the call which you can use to enrich your own data. The impression event will contain the following data, including the entire context of the call (which will expand if you provide more values to the unleash context): 
+
+```
```
Comment (thomasheartman) [inline website/docs/advanced/impression_data.md:63]:
```suggestion
```js
```

File: website/docs/advanced/impression_data.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+This step assumes that you have set up an application connected to an unleash instance using one of our [SDKs](/sdks)
+
+:::caution 
+Currently this functionality is only supported in [unleash-proxy-client](/sdks/proxy-javascript) and [proxy-client-react](/sdks/proxy-react)
+:::
+
+Once you have set up your client application with the SDK you can listen to impression events on the client. Example with unleash-proxy-client:
+
+```
```
Comment (thomasheartman) [inline website/docs/advanced/impression_data.md:42]:
```suggestion
```js
```

File: website/docs/advanced/impression_data.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+---
+id: impression_data
+title: Impression data
+---
+
+Impression data represents that unleash uses locally in order to resolve a feature toggle to true or false. This data does not leave your application, and the data is not sent to unleash - but it is used for the purposes of resolving feature toggles.
```
Comment (thomasheartman) [inline website/docs/advanced/impression_data.md:6]: I think you might have forgotten a word or two in the first sentence there 😅 What does impression data represent?

File: website/sidebars.js [hunk 4]
```diff
@@ -246,6 +246,7 @@ module.exports = {
                         'user_guide/activation_strategy',
                         'advanced/archived_toggles',
                         'advanced/audit_log',
+                        'advanced/impression_data',
```
Comment (thomasheartman) [inline website/sidebars.js:249]: We should move this to between projects and role-based access control to keep the list sorted 🤓

**Review by thomasheartman — COMMENTED** [review]
Great work 🙌🏼  I think this is a really good starting point and that it covers a lot of the most important parts of the feature. I think it could benefit from some work on the structure, though.

Personally, I'd like to see a clearer description of what impression data is, what form it takes (e.g. what data does it carry), and how you use it. 

Outlined the way it is now, it's more like a how-to guide (which is great, but not the same as [reference documentation as defined by diataxis](https://diataxis.fr/reference/)). I can see a single paragraph on "how to enable it" being useful though.

That said, I'm very happy to discuss this, and I also wouldn't mind taking a stab at reorganizing it a bit if you'd want?

**Review by ivarconr — APPROVED** [review]

File: website/docs/advanced/impression_data.md (hunk 3 above)
Comment (FredrikOseberg) [reply website/docs/advanced/impression_data.md:6]: Woops! Yes that is totally correct.

File: website/sidebars.js (hunk 4 above)
Comment (FredrikOseberg) [reply website/sidebars.js:249]: Fair!

**Comment by FredrikOseberg** [comment]
> Great work 🙌🏼 I think this is a really good starting point and that it covers a lot of the most important parts of the feature. I think it could benefit from some work on the structure, though.
> 
> Personally, I'd like to see a clearer description of what impression data is, what form it takes (e.g. what data does it carry), and how you use it.
> 
> Outlined the way it is now, it's more like a how-to guide (which is great, but not the same as [reference documentation as defined by diataxis](https://diataxis.fr/reference/)). I can see a single paragraph on "how to enable it" being useful though.
> 
> That said, I'm very happy to discuss this, and I also wouldn't mind taking a stab at reorganizing it a bit if you'd want?
> 
> ## Edit
> Oh, and I chucked in a couple minor suggestions that jumped out at me regarding syntax highlighting, documentation ordering, and a sentence potentially missing a word 🤓

Thanks for the feedback! If you want to take a stab at reorganizing it that would fantastic.

**Review by thomasheartman — APPROVED** [review]
Looks good to me! 🚂 Thanks for putting this together 🙌🏼

**Pull request merged**